osquery/filesystem/windows/fileops.cpp

```diff
@@ -1082,6 +1082,79 @@ size_t PlatformFile::size() const {
   return ::GetFileSize(handle_, nullptr);
 }
 
+bool platformSetSafeDbPerms(const std::string& path) {
+  unsigned long sid_size = SECURITY_MAX_SID_SIZE;
+  std::vector<char> admins_buf(sid_size);
+  PSID admins = static_cast<PSID>(admins_buf.data());
+  if (!::CreateWellKnownSid(
+          WinBuiltinAdministratorsSid, nullptr, admins, &sid_size)) {
+    return false;
+  }
+
+  std::vector<char> system_buf(sid_size);
+  PSID system = static_cast<PSID>(system_buf.data());
+  if (!::CreateWellKnownSid(WinLocalSystemSid, nullptr, system, &sid_size)) {
+    return false;
+  }
+
+  EXPLICIT_ACCESS ea;
+  ea.grfAccessMode = SET_ACCESS;
+  ea.grfAccessPermissions = GENERIC_ALL;
+  ea.grfInheritance = SUB_CONTAINERS_AND_OBJECTS_INHERIT;
+
+  PTRUSTEE_A trust = static_cast<PTRUSTEE_A>(malloc(sizeof(TRUSTEE_A)));
```

Comment:
We could use same pattern of vector data usage, and you could get rid of frees. But things don't look to be leaking.

Reply:
Eh. I'm happy to have a follow up issue to clean up the `fileops` code, as I mentioned in the description there's likely soon to be some dead code floating around here, but if it's alright I'd rather just accept the `malloc` so long as it's not leaking. Lemme know, I'm happy to rewrite if we'd like :)

Reply:
Naah. It's cool 👍